ListVolumes : Panic detected on v4.4.1

First, thanks for this great project.

Panic detected in my dev environment.

I'm developing a CSI driver, and I can see a panic during the `ListVolumes`.
Do you have any idea where it can come from, if it's can come from my implementation of the CSI proto?

In the List request of the Controller, proto, I'm giving this:
- `csi.Volume`
    - `VolumeId`
    - `CapacityBytes` (not negative)
    - `AccessibleTopology`

My driver return well `PluginCapability_Service_VOLUME_ACCESSIBILITY_CONSTRAINTS`

I'm using this Image: `registry.k8s.io/sig-storage/csi-attacher:v4.4.1`
```yaml
        - name: csi-attacher
          image: registry.k8s.io/sig-storage/csi-attacher:v4.4.1
          args:
            - "--v=5"
            - "--csi-address=$(CSI_ADDRESS)"
            - "--leader-election"
```

I'm running k8s `v1.27.7` with callico `v3.26.3`

```Golang
I1020 16:16:06.226090       1 csi_handler.go:123] Reconciling VolumeAttachments with driver backend state
I1020 16:16:06.226120       1 connection.go:193] GRPC call: /csi.v1.Controller/ListVolumes
I1020 16:16:06.226155       1 connection.go:194] GRPC request: {}
I1020 16:16:06.307062       1 connection.go:200] GRPC response: {"entries":[{},{"volume":{"accessible_topology":[{"segments":{"topology.csi.exoscale.com/zone":"at-vie-1"}}],"capacity_bytes":214748364800,"volume_id":"XXXXXXXXXXX"}}]}
I1020 16:16:06.307326       1 connection.go:201] GRPC error: <nil>
E1020 16:16:06.307634       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 103 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1947c00?, 0x2b86190})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x1947c00, 0x2b86190})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/kubernetes-csi/external-attacher/pkg/attacher.(*CSIVolumeLister).ListVolumes(0xc0002cd3f8, {0x1e4e0f0, 0xc0002e54d0})
	/workspace/pkg/attacher/lister.go:55 +0x1a3
github.com/kubernetes-csi/external-attacher/pkg/controller.(*csiHandler).ReconcileVA(0xc0002f4f70)
	/workspace/pkg/controller/csi_handler.go:134 +0x16b
github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run.func1()
	/workspace/pkg/controller/controller.go:144 +0x2a
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0004f7f00?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0000d4e10?, {0x1e347e0, 0xc00032c510}, 0x1, 0xc000490840)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0000d4e1a?, 0xdf8475800, 0x0, 0x1?, 0xc0000d4e1e?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x1?, 0xc0000d4e22?, 0x1?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161 +0x25
created by github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run
	/workspace/pkg/controller/controller.go:143 +0x49f
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x174a2c3]

goroutine 103 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x1947c00, 0x2b86190})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/kubernetes-csi/external-attacher/pkg/attacher.(*CSIVolumeLister).ListVolumes(0xc0002cd3f8, {0x1e4e0f0, 0xc0002e54d0})
	/workspace/pkg/attacher/lister.go:55 +0x1a3
github.com/kubernetes-csi/external-attacher/pkg/controller.(*csiHandler).ReconcileVA(0xc0002f4f70)
	/workspace/pkg/controller/csi_handler.go:134 +0x16b
github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run.func1()
	/workspace/pkg/controller/controller.go:144 +0x2a
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0004f7f00?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0000d4e10?, {0x1e347e0, 0xc00032c510}, 0x1, 0xc000490840)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0000d4e1a?, 0xdf8475800, 0x0, 0x1?, 0xc0000d4e1e?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x1?, 0xc0000d4e22?, 0x1?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161 +0x25
created by github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run
	/workspace/pkg/controller/controller.go:143 +0x49f
```

Thank you

I fixed the empty `entrie`:

But still a panic, I'm still investigating.

```Golang
I1024 10:16:44.001049       1 connection.go:200] GRPC response: {"entries":[{"volume":{"accessible_topology":[{"segments":{"topology.csi.exoscale.com/zone":"at-vie-1"}}],"capacity_bytes":214748364800,"volume_id":"90a423f8-5e0a-4e52-b5a3-4f50c27c64fd"}}]}
I1024 10:16:44.001174       1 connection.go:201] GRPC error: <nil>
E1024 10:16:44.001392       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 105 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1947c00?, 0x2b86190})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x1947c00, 0x2b86190})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/kubernetes-csi/external-attacher/pkg/attacher.(*CSIVolumeLister).ListVolumes(0xc0003c4528, {0x1e4e0f0, 0xc0005aac60})
	/workspace/pkg/attacher/lister.go:55 +0x1ab
github.com/kubernetes-csi/external-attacher/pkg/controller.(*csiHandler).ReconcileVA(0xc000111d40)
	/workspace/pkg/controller/csi_handler.go:134 +0x16b
github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run.func1()
	/workspace/pkg/controller/controller.go:144 +0x2a
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x1e347e0, 0xc0005aac30}, 0x1, 0xc000101d40)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0xdf8475800, 0x0, 0x0?, 0x0?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161 +0x25
created by github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run
	/workspace/pkg/controller/controller.go:143 +0x49f
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x174a2cb]

goroutine 105 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x1947c00, 0x2b86190})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/kubernetes-csi/external-attacher/pkg/attacher.(*CSIVolumeLister).ListVolumes(0xc0003c4528, {0x1e4e0f0, 0xc0005aac60})
	/workspace/pkg/attacher/lister.go:55 +0x1ab
github.com/kubernetes-csi/external-attacher/pkg/controller.(*csiHandler).ReconcileVA(0xc000111d40)
	/workspace/pkg/controller/csi_handler.go:134 +0x16b
github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run.func1()
	/workspace/pkg/controller/controller.go:144 +0x2a
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x1e347e0, 0xc0005aac30}, 0x1, 0xc000101d40)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0xdf8475800, 0x0, 0x0?, 0x0?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161 +0x25
created by github.com/kubernetes-csi/external-attacher/pkg/controller.(*CSIAttachController).Run
	/workspace/pkg/controller/controller.go:143 +0x49f
```

--- a/pkg/attacher/lister.go
+++ b/pkg/attacher/lister.go
@@ -52,7 +52,11 @@ func (a *CSIVolumeLister) ListVolumes(ctx context.Context) (map[string]([]string
 		}
 
 		for _, e := range rsp.Entries {
-			p[e.GetVolume().VolumeId] = e.Status.PublishedNodeIds
+			if e.GetVolume() == nil || e.GetStatus() == nil {
+				continue
+			}
+
+			p[e.GetVolume().VolumeId] = e.GetStatus().GetPublishedNodeIds()
 		}
 		tok = rsp.NextToken
 

